pkg/adaptor/cloud/cloud.go

```diff
@@ -75,6 +78,14 @@ type sandbox struct {
 	instanceName string
 	instanceID   string
 	netNSPath    string
+	instanceType string
+}
+
+type InstanceTypeTuple struct {
+	InstanceType string
+	Vcpu         int64
+	// In MiB
+	Memory int64
```

Comment:
We can possiblyuse the `InstanceSpec` type here like this.

```
type InstanceType struct {
	Name string
	Spec InstanceSpec
}
```

Reply:
The naming is an issue though. Some cloud providers (eg. aws) use InstanceType as the name/flavour and hence I kept instancetype as the name and InstanceTypeTuple as the detailed spec

How about using MachineType and MachineTypeSpec to be generic ?
```
type MachineType struct {
     Name string
     Spec MachineTypeSpec
}

type MachineTypeSpec struct {
      Name string
      Vcpus int64
      Memory int64
}
```

Reply:
Could you show some example data for each struct? I thought machine type is an architecture string like x86_64, s390x, ..., but it looks like something else.

Reply:
I used MachineType to be in sync with the kata annotation of machine_type which we are using for instance type name. For qemu it's the qemu machine type like q35, microvm etc. For remote hypervisor it becomes the instance types (names like t2.small etc). We can use arch to indicate the architecture.

Reply:
Example data
```
MachineType {
   Name: "t2.small",
   Spec: MachineTypeSpec{
        Name:   "t2.small",
        Vcpus:  1,
        Memory: 2048,
    },
}
```

Reply:
@yoheiueda does this make sense, or you prefer InstanceType?

Reply:
On rethinking, I do feel machinetype can be confusing with arch. 
So may be we can use InstanceTypeSpec 
```
type InstanceTypeSpec struct {
	InstanceType string
	VCPUs     int
	Memory   int
	Arch         string
	GPUs       int
}
```

Reply:
It looks good to me.